src/app.rs

```diff
@@ -176,76 +154,85 @@ impl<Ms, Mdl, INodes: IntoNodes<Ms> + 'static, GMs: 'static> App<Ms, Mdl, INodes
-                }
-            }) as Box<dyn IntoAfterMount<Ms, Mdl, INodes, GMs>>,
+        let app = Self {
+            cfg: Rc::new(AppCfg::new(
+                util::window().document().expect("get window's document"),
+                // @TODO_B: trim out to remove mount_point field
+                root_element.get_element().expect("get root element"),
+                update,
+                view,
+                base_path,
+            )),
+            data: Rc::new(AppData {
+                model: RefCell::new(None),
+                main_el_vdom: RefCell::new(None),
+                popstate_closure: RefCell::new(None),
+                hashchange_closure: RefCell::new(None),
+                window_event_handler_manager: RefCell::new(EventHandlerManager::new()),
+                sub_manager: RefCell::new(SubManager::new()),
+                msg_listeners: RefCell::new(Vec::new()),
+                scheduled_render_handle: RefCell::new(None),
+                after_next_render_callbacks: RefCell::new(Vec::new()),
+                render_info: Cell::new(None),
+            }),
         };
-        let app = Self::new(
-            update,
-            None,
-            view,
-            root_element,
-            None,
-            None,
-            Some(app_init_cfg),
-            base_path,
-        );
-        app.run()
-    }
 
-    /// Creates a new `AppBuilder` instance. It's the standard way to create a Seed app.
-    ///
-    /// Then you can call optional builder methods like `routes` or `sink`.
-    /// And you have to call method `build_and_start` to build and run a new `App` instance.
-    ///
-    /// _NOTE:_ If your `Model` doesn't implement `Default`, you have to call builder method `after_mount`.
-    ///
-    /// # Example
-    ///
-    /// ```rust,no_run
-    ///fn update(msg: Msg, model: &mut Model, _orders: &mut impl Orders<Msg, GMsg>) {
-    ///   match msg {
-    ///       Msg::Clicked => model.clicks += 1,
-    ///   }
-    ///}
-    ///
-    ///fn view(model: &Model) -> impl IntoNodes<Msg> {
-    ///   vec![
-    ///       button![
-    ///           format!("Clicked: {}", model.clicks),
-    ///           simple_ev(Ev::Click, Msg::Clicked),
-    ///       ],
-    ///   ]
-    ///}
-    ///
-    ///App::builder(update, view)
-    /// ```
-    pub fn builder(
-        update: UpdateFn<Ms, Mdl, INodes, GMs>,
-        view: ViewFn<Mdl, INodes>,
-    ) -> AppBuilder<Ms, Mdl, INodes, GMs, UndefinedInitAPI> {
-        // @TODO: Remove as soon as Webkit is fixed and older browsers are no longer in use.
-        // https://github.com/David-OConnor/seed/issues/241
-        // https://bugs.webkit.org/show_bug.cgi?id=202881
-        let _ = util::document().query_selector("html");
+        app.data
+            // @TODO_B we are trying to remove/rename this field
+            .main_el_vdom
+            // @TODO_B with the removal of MountType, this becomes argument-free. This needs looking at
```

Comment:
You mentioned earlier about the `mount_point` field in cfg.  If rendering the vdom doesn't require the mount point, you can isolate it here and pass it directly to `bootstrap_vdom`. Otherwise, `mount_point` has to stay. Made another comment in `rerender_vdom`. I think it's the only place `mount_point` is used, and I'm pretty sure it needs to be kept around one way or another.

Reply:
*used as in, if it were removed, something would break here. Just to be clear.